test/system/include/pattern/boost_regex.hpp

```diff
@@ -0,0 +1,31 @@
+#ifndef METASHELL_SYSTEM_TEST_INCLUDE_BOOST_REGEX_HPP
+#define METASHELL_SYSTEM_TEST_INCLUDE_BOOST_REGEX_HPP
+
+// Metashell - Interactive C++ template metaprogramming shell
+// Copyright (C) 2016, Abel Sinkovics (abel@sinkovics.hu)
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#if defined(__GNUC__) && !defined(__clang__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
Strange. Boost is included with `-isystem` which should silence all warnings from those headers. Seems like a minor compiler bug. Anyway, nice workaround :+1: